### PR TITLE
Add pekko and pekko-http modules

### PR DESCRIPTION
**Pull request by JD557**
Adds `pekko` and `pekko-http` modules, with support for Pekko 1.0.2 and Pekko-HTTP 1.0.1.

They are pretty much a copy of the `akka`/`akka-http` modules.

### CODE REVIEW

File: modules/pekko-http/build.sbt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "org.apache.pekko" %% "pekko-stream" % "1.0.2" % "provided",
+  "org.apache.pekko" %% "pekko-http" % "1.0.1"
+)
+mdocLibraryDependencies ++= Seq(
+  "org.apache.pekko" %% "pekko-stream" % "1.0.2"
+)
+
+developers := List(
+  Developer("himanshu4141", "Himanshu Yadav", "himanshu4141@gmail.com", url("https://github.com/himanshu4141"))
+)
```
Comment (jcazevedo) [inline modules/pekko-http/build.sbt]: Not really. You can omit this if you don't want to put your name in.

File: modules/pekko-http/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Pekko-Http module for PureConfig
```
Comment (jcazevedo) [inline modules/pekko-http/docs/README.md]: Nitpick: Can we spell this the same way [the project does](https://pekko.apache.org/docs/pekko-http/current/), i.e. "Pekko HTTP"?

```suggestion
# Pekko HTTP module for PureConfig
```

File: modules/pekko-http/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Adds support for [Pekko-Http](https://pekko.apache.org/docs/pekko-http/current/common/http-model.html)'s Uri class to PureConfig. PRs adding support
```
Comment (jcazevedo) [inline modules/pekko-http/docs/README.md:3]: Nitpick: Same as above.

```suggestion
Adds support for [Pekko HTTP](https://pekko.apache.org/docs/pekko-http/current/common/http-model.html)'s Uri class to PureConfig. PRs adding support
```

File: modules/pekko-http/docs/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+```scala
+libraryDependencies += "com.github.pureconfig" %% "pureconfig-pekko-http" % "@VERSION@"
+```
+
+## Example
+
+To load an `Uri` into a configuration, create a class to hold it:
+
+```scala mdoc:silent
+import pekko.http.scaladsl.model.Uri
```
Comment (jcazevedo) [inline modules/pekko-http/docs/README.md]: Our CI step that checks for mdoc output is broken.

```suggestion
import org.apache.pekko.http.scaladsl.model.Uri
```

File: modules/pekko-http/build.sbt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import Dependencies.Version._
+
+crossScalaVersions := Seq(scala212, scala213)
+
+libraryDependencies ++= Seq(
+  "org.apache.pekko" %% "pekko-stream" % "1.0.2" % "provided",
```
Comment (jcazevedo) [inline modules/pekko-http/build.sbt]: I think only pekko-actor is required here for `ActorRefFactory`.

```suggestion
  "org.apache.pekko" %% "pekko-actor" % "1.0.2" % "provided",
```

File: modules/pekko-http/build.sbt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  "org.apache.pekko" %% "pekko-http" % "1.0.1"
+)
+mdocLibraryDependencies ++= Seq(
+  "org.apache.pekko" %% "pekko-stream" % "1.0.2"
```
Comment (jcazevedo) [inline modules/pekko-http/build.sbt]:
```suggestion
  "org.apache.pekko" %% "pekko-actor" % "1.0.2"
```

File: modules/pekko/build.sbt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import Dependencies.Version._
+
+crossScalaVersions := Seq(scala212, scala213)
+
+libraryDependencies ++= Seq("org.apache.pekko" %% "pekko-actor" % "1.0.2")
+
+developers := List(
+  Developer("derekmorr", "Derek Morr", "morr.derek@gmail.com", url("https://github.com/derekmorr")),
+  Developer("ruippeixotog", "Rui Gonçalves", "ruippeixotog@gmail.com", url("https://github.com/ruippeixotog"))
+)
```
Comment (jcazevedo) [inline modules/pekko/build.sbt]: Also omit this.

File: modules/pekko-http/README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+Adds support for [Pekko-Http](https://pekko.apache.org/docs/pekko-http/current/common/http-model.html)'s Uri class to PureConfig. PRs adding support
+for other classes are welcome :)
+
+## Add pureconfig-pekko-http to your project
+
+In addition to [core PureConfig](https://github.com/pureconfig/pureconfig), you'll need:
+
+```scala
+libraryDependencies += "com.github.pureconfig" %% "pureconfig-pekko-http" % "@VERSION@"
```
Comment (jcazevedo) [inline modules/pekko-http/README.md]: Running `pekko-http-docs/mdoc` should have this replaced.

```suggestion
libraryDependencies += "com.github.pureconfig" %% "pureconfig-pekko-http" % "0.17.5"
```

File: modules/pekko/docs/README.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
+# Pekko module for PureConfig
+
+Adds support for selected [Pekko](https://pekko.apache.org/) classes to PureConfig.
+
+## Add pureconfig-pekko to your project
+
+In addition to [core PureConfig](https://github.com/pureconfig/pureconfig), you'll need:
+
+```scala
+libraryDependencies += "com.github.pureconfig" %% "pureconfig-pekko" % "@VERSION@"
```
Comment (jcazevedo) [inline modules/pekko/docs/README.md:10]: Running `pekko-http-docs/mdoc` should have this replaced.

```suggestion
libraryDependencies += "com.github.pureconfig" %% "pureconfig-pekko" % "0.17.5"
```

**Review by jcazevedo — CHANGES_REQUESTED** [review]
Thanks for this!

File: modules/pekko-http/build.sbt (hunk 1 above)
Comment (JD557) [reply modules/pekko-http/build.sbt]: Removed in 12f0924

File: modules/pekko/build.sbt (hunk 7 above)
Comment (JD557) [reply modules/pekko/build.sbt]: Removed in 12f0924

File: modules/pekko-http/docs/README.md (hunk 3 above)
Comment (JD557) [reply modules/pekko-http/docs/README.md:3]: Fixed in 8655d5b

File: modules/pekko-http/docs/README.md (hunk 4 above)
Comment (JD557) [reply modules/pekko-http/docs/README.md]: Oops, nice catch.
Fixed in 8655d5b

File: modules/pekko-http/README.md (hunk 8 above)
Comment (JD557) [reply modules/pekko-http/README.md]: My bad, I copy pasted those lines and forgot to run the command.
Regenerated the files in 8655d5b

File: modules/pekko/docs/README.md (hunk 9 above)
Comment (JD557) [reply modules/pekko/docs/README.md:10]: Same as above.
Regenerated the files in 8655d5b

File: modules/pekko-http/build.sbt (hunk 5 above)
Comment (JD557) [reply modules/pekko-http/build.sbt]: That seems to work. Changed in f022109

File: modules/pekko-http/build.sbt (hunk 6 above)
Comment (JD557) [reply modules/pekko-http/build.sbt]: Changed in f022109

**Review by jcazevedo — APPROVED** [review]
Thanks!

**Pull request merged**